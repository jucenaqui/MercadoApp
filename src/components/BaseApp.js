import React, { useEffect, useState } from 'react';
import queryString from 'query-string';

import ProductApp from './product/ProductApp';
import { getProducts } from '../hooks/GetProducts';
import { HeaderBarApp } from './header/HeaderBarApp';

const BaseApp = ( { history, location } ) => {

    const { search = '' } = queryString.parse( location.search );

    const [ products, setProducts ] = useState({
        items: [],
        search
    });

    const { items } = products;

    useEffect(()=> {
        getProducts(search)
        .then( (items) => {
            setProducts( value => { 
                return {
                    ...value,
                    items
                }
            });
        });
    }, [search ])

    return (
        <>
            <HeaderBarApp setProducts={setProducts} history={ history } location={ location } />
            {
                items && <ProductApp products= { items } />
            }
        </>
    )
}

export default BaseApp;