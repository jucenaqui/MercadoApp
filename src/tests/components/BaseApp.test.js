import { shallow } from "enzyme";
import BaseApp from "../../components/BaseApp";
import { createLocation } from 'history';

describe('Test to <App />', () => {

    const location = createLocation();
    let wrapper = shallow(<BaseApp location={location} />);

    test('should show the component succesfull', () => {
        expect( wrapper ).toMatchSnapshot();
    });

});