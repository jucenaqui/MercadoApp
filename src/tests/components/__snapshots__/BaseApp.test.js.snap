// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to <App /> should show the component succesfull 1`] = `<Fragment />`;
